ocaml/idl/datamodel.ml

```diff
@@ -7732,6 +7732,124 @@ module VUSB = struct
+          )
+        ]
+      ~allowed_roles:_R_POOL_ADMIN ()
+
+  let set_components =
+    call ~name:"set_components" ~in_oss_since:None ~lifecycle
+      ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+      ~params:
+        [
+          (Ref _tracing, "self", "The TracerProvider")
+        ; ( Set String
+          , "components"
+          , "The components the TracerProvider is to be set to"
+          )
+        ]
+      ~allowed_roles:_R_POOL_ADMIN ()
+
+  let set_filters =
+    call ~name:"set_filters" ~in_oss_since:None ~lifecycle
+      ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+      ~params:
+        [
+          (Ref _tracing, "self", "The TracerProvider")
+        ; ( Set String
+          , "filters"
+          , "The filter the TracerProvider is to be set to"
+          )
+        ]
+      ~allowed_roles:_R_POOL_ADMIN ()
+
+  let set_processors =
+    call ~name:"set_processors" ~in_oss_since:None ~lifecycle
+      ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+      ~params:
+        [
+          (Ref _tracing, "self", "The TracerProvider")
+        ; ( Set String
+          , "processors"
+          , "The processors the TracerProvider is to be set to"
+          )
+        ]
+      ~allowed_roles:_R_POOL_ADMIN ()
+
+  let t =
+    create_obj ~name:_tracing ~descr:"Describes a TracerProvider"
+      ~doccomments:[] ~gen_constructor_destructor:false ~gen_events:true
+      ~in_db:true ~lifecycle ~persist:PersistEverything ~in_oss_since:None
+      ~messages_default_allowed_roles:_R_POOL_ADMIN
+      ~contents:
+        ([uid _tracing ~lifecycle]
+        @ [
+            field ~qualifier:DynamicRO ~ty:(Set (Ref _host)) ~lifecycle "hosts"
+              "A list of hosts the TracerProvider is to be registered on"
+              ~ignore_foreign_key:true
+          ; field ~qualifier:DynamicRO ~ty:String ~lifecycle "name_label"
+              "A user-assinged label for a TracerProvider"
+          ; field ~qualifier:DynamicRO
+              ~ty:(Map (String, String))
+              ~lifecycle "tags"
+              "Tags to be set on spans to provide extra information"
+          ; field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle "endpoints"
+              "Endpoints where spans are exported to"
+          ; field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle "components"
+              "Componenets the TraverProvider is to be registered on"
+          ; field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle "filters"
+              "Filters for operations that spans will be produced for"
+          ; field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle "processors"
+              "Processors that will filter/map spans before being exported"
+          ; field ~qualifier:DynamicRO ~ty:Bool ~lifecycle "status"
+              "The status of the TracerProvider"
+          ]
+        )
+      ~messages:
+        [
+          set_status
```

Comment:
There is no constructor at all?

Reply:
You somewhere to start though. Are you pre-creating a Tracing object somewhere? I thought we'd have no tracing by default, and then if you want tracing, you create a Tracing object with the relevant parameters such as the endpoint etc.